services/QuillLMS/spec/queries/snapshots/top_concepts_assigned_query_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Snapshots
+  describe TopConceptsAssignedQuery do
+    include_context 'Snapshots Period CTE'
+
+    context 'big_query_snapshot', :big_query_snapshot do
+      let(:num_classrooms) { 1 }
+      let(:num_concepts) { 11 }
+
+      let(:concepts) { create_list(:concept, num_concepts) }
+      let(:activities) { concepts.map { |concept| create(:activity, data: {modelConceptUID: concept.uid}) } }
+      # Note that we're setting assigned_student_ids to an arbitrary one-length array because we don't actually need to reference the students in question, so any number "works" here.
+      let(:classroom_units) { classrooms.map { |classroom| create_list(:classroom_unit, num_concepts, classroom: classroom, assigned_student_ids: [1]) }.flatten }
+
+      # We have one activity connected to each concept.
+      # We have a number of classroom_units equal to the number of concepts.
+      # We want to assign one concept to all classroom_units, the next concept to all but one, the next to all but two, and so on so that each concept is used a different number of times.
+      let(:unit_activity_bundles) do
+        classroom_units.map.with_index do |classroom_unit, classroom_unit_index|
+          Array(0..(num_concepts - (1 + classroom_unit_index))).map do |target_activity_index|
+            create(:unit_activity, activity: activities[target_activity_index], unit: classroom_unit.unit)
+          end
+        end
+      end
+
+      let(:runner_context) {
+        [
+          classrooms,
+          teachers,
+          classrooms_teachers,
+          schools,
+          schools_users,
+          classroom_units,
+          concepts,
+          activities,
+          classroom_units
+        ]
+      }
+
+      let(:cte_records) { [runner_context, unit_activity_bundles] }
+
+      let(:runner) { QuillBigQuery::TestRunner.new(cte_records) }
```

Comment:
`let(:runner)` and `let(:results)` can be removed since they are already defined in the shared_context [here](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/spec/support/shared_contexts/quill_big_query_test_runner_setup.rb#L7)